modules/src/ics02_client/state.rs

```diff
@@ -18,3 +19,24 @@ pub trait ConsensusState {
     /// Performs basic validation of the consensus state
     fn validate_basic(&self) -> Result<(), Self::ValidationError>;
 }
+
+pub trait ClientState {
+    type ValidationError: std::error::Error;
+
+    /// Client ID of this state
+    fn client_id(&self) -> ClientId;
+
+    /// Type of client associated with this state (eg. Tendermint)
+    fn client_type(&self) -> ClientType;
+
+    /// Height of consensus state
+    fn get_latest_height(&self) -> Height;
+
+    /// Freeze status of the client
+    fn is_frozen(&self) -> bool;
+
+    fn verify_client_consensus_state(
+        &self,
+        root: &CommitmentRoot,
+    ) -> Result<(), Self::ValidationError>;
```

Comment:
I guess the other state verification functions are not included on purpose?
https://github.com/cosmos/cosmos-sdk/blob/b5a658729170e577b0e6e809e73d5e0756b90e07/x/ibc/02-client/exported/exported.go#L26-L92

Reply:
Yes, a bit more work to get there :)